baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/DangerousParallelStreamUsage.java

```diff
@@ -0,0 +1,72 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.Category;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "DangerousParallelStreamUsage",
+        category = Category.JDK,
+        severity = SeverityLevel.WARNING,
+        summary = "Discourage usage of .parallel() in Java streams.")
+public final class DangerousParallelStreamUsage extends BugChecker implements BugChecker.MethodInvocationTreeMatcher {
+    private static final long serialVersionUID = 1L;
+    private static final String MORE_STREAMS_URL = "https://github.com/palantir/streams/"
+            + "blob/1.9.1/src/main/java/com/palantir/common/streams/MoreStreams.java#L53";
+    private static final String ERROR_MESSAGE = "Should not use .parallel() on a Java stream. "
+            + "Doing so has very subtle and potentially severe performance implications, so in general you're better "
+            + "off using " + MORE_STREAMS_URL + " which allows you to provide your own executor service and "
+            + "specify the desired parallelism (i.e. the max number of concurrent tasks that will be submitted).\n"
+            + "On the contrary the implementation of Java parallel streams uses a globally shared ForkJoinPool and "
+            + "does not allow you to provide your own pool. Fork/join pools implement work-stealing, where any thread "
+            + "might steal a task from a different thread's queue when blocked waiting for a subtask to complete. "
+            + "This might not seem like an issue at first glance, but if you use .parallel() for short tasks "
+            + "extensively throughout your codebase and later on you add one piece of code that uses .parallel() "
+            + "for long (e.g. I/O) tasks, the other parts of your codebase that use .parallel(), and that you'd "
+            + "expect to have consistent performance, might experience performance degradation for no apparent reason. "
+            + "The reason is work stealing.\n"
+            + "You can suppress this warning if you are certain that all your code will always only use .parallel() "
+            + "for short tasks, but even then, you have no real control over the level of parallelism, so you're "
+            + "still better off using MoreStreams (linked above)\n"
```

Comment:
nit: Not sure if specifying all of this background is necessary.

Reply:
Do you think this much background is confusing? Any suggestion on what to cut? In general I dislike error prone checks that tell me not to do something without explanation. Happy to defer that to the stackoverflow link though

Reply:
Cool, if you think it gives useful detail that's fine.

Reply:
I think its useful to have all the detail, it gives users more context being the rule and extra motivation for fixing the issue